### PR TITLE
feat(button link): add href prop to the link button render as an anchor

### PR DESCRIPTION
**Pull request by dehmirandac2**
## Motivation
To fix the [issue 418](https://github.com/Gympass/yoga/issues/418) I'm adding a new prop to the `Button.Link` component.
If you add the `href` prop to the component it will render as an anchor.

Example:
```js
<Button.Link href="http://www.google.com" target="blank">Open external url</Button.Link>
```

How the component will be rendered:
```js
<a href="http://www.google.com" target="blank">Open external url</a>
```

### Default value is an anchor
Screen Shot 2022-04-06 at 09 36 25

### To render as a button, you should add the "as"
Screen Shot 2022-04-06 at 09 36 32

### Table of props
Screen Shot 2022-04-06 at 09 36 39

## Changes
- Using [styled component "as" ](https://styled-components.com/docs/api#as-polymorphic-prop)to render the the component as an anchor element by default
- If the user want to use as button, they should add `as="button"`
- Add snapshot test
- Add more info to the documentation

### CODE REVIEW

**Review by ericcleao — APPROVED** [review]

File: packages/doc/content/components/components/button/link-web.mdx [hunk 1]
```diff
@@ -7,6 +7,16 @@
 </Box>
 ```
 
+#### Using as an anchor
```
Comment (Wendlereis) [inline packages/doc/content/components/components/button/link-web.mdx:10]: Would be nice to push this as recommended usage for this component. How could we create a deprecation strategy here?

cc: @mmartinsoliv @tcK1 @dehmirandac2

**Review by frgiovanna — APPROVED** [review]

**Comment by Wendlereis** [comment]
I have noticed that we have 2 issues for this PR. One of them has a suggestion to make any button behaves as a `<a/>` given a `href` prop. 

Do you think worth add this on this PR?

**Comment by dehmirandac2** [comment]
> I have noticed that we have 2 issues for this PR. One of them has a suggestion to make any button behaves as a `<a/>` given a `href` prop.
> 
> Do you think worth add this on this PR?

@Wendlereis I'm extending this functionality to all Buttons [in this PR](https://github.com/Gympass/yoga/pull/449). If you could check, please

**Comment by mmartinsoliv** [comment]
Hi, @dehmirandac2 thank you so much for your contribution! 

I'm thinking about this PR here and what do you think about us keeping the `Button.Link` with the `a` tag as the default?

For the other Buttons, we can keep the `button` tag as default, if we need the default `Button` to be with the `a` tag, we can use the polymorphic prop.

`styled-components` API example: [doc](https://styled-components.com/docs/api#as-polymorphic-prop).

What do you think?

**Pull request closed without merging**